torchgeo/datasets/globbiomass.py

```diff
@@ -0,0 +1,282 @@
+        "N80E100_gsv.zip": "73067e0fac442c330ae2294996280042",
+        "N80E140_agb.zip": "1b51ce0df0dba925c5ef2149bebca115",
+        "N80E140_gsv.zip": "37ee3047d281fc34fa3a9e024a8317a1",
+        "N80W020_agb.zip": "60dde6adc0dfa219a34c976367f571c0",
+        "N80W020_gsv.zip": "b7be4e97bb4179710291ee8dee27f538",
+        "N80W060_agb.zip": "db7d35d0375851c4a181c3a8fa8b480e",
+        "N80W060_gsv.zip": "d36ffcf4622348382454c979baf53234",
+        "N80W100_agb.zip": "c0dbf53e635dabf9a4d7d1756adeda69",
+        "N80W100_gsv.zip": "abdeaf0d65da1216c326b6d0ce27d61b",
+        "N80W140_agb.zip": "7719c0efd23cd86215fea0285fd0ea4a",
+        "N80W140_gsv.zip": "499969bed381197ee9427a2e3f455a2e",
+        "N80W180_agb.zip": "e3a163d1944e1989a07225d262a01c6f",
+        "N80W180_gsv.zip": "5d39ec0368cfe63c40c66d61ae07f577",
+        "S40E140_agb.zip": "263eb077a984117b41cc7cfa0c32915b",
+        "S40E140_gsv.zip": "e0ffad85fbade4fb711cc5b3c7543898",
+        "S40W060_agb.zip": "2cbf6858c48f36add896db660826829b",
+        "S40W060_gsv.zip": "04dbfd4aca0bd2a2a7d8f563c8659252",
+        "S40W100_agb.zip": "ae89f021e7d9c2afea433878f77d1dd6",
+        "S40W100_gsv.zip": "b6aa3f276e1b51dade803a71df2acde6",
+    }
+
+    def __init__(
+        self,
+        root: str = "data",
+        crs: Optional[CRS] = None,
+        res: Optional[float] = None,
+        measurement: str = "agb",
+        transforms: Optional[Callable[[Dict[str, Any]], Dict[str, Any]]] = None,
+        cache: bool = True,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize a new Dataset instance.
+
+        Args:
+            root: root directory where dataset can be found
+            crs: :term:`coordinate reference system (CRS)` to warp to
+                (defaults to the CRS of the first file found)
+            res: resolution of the dataset in units of CRS
+                (defaults to the resolution of the first file found)
+            measurement: use data from 'agb' or 'gsv' measurement
+            transforms: a function/transform that takes an input sample
+                and returns a transformed version
+            cache: if True, cache file handle to speed up repeated sampling
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            FileNotFoundError: if no files are found in ``root``
+            RuntimeError: if dataset is missing or checksum fails
+            AssertionError: if measurement argument is invalid, or not a str
+        """
+        self.root = root
+        self.checksum = checksum
+
+        assert isinstance(measurement, str), "Measurement argument must be a str."
+        assert (
+            measurement in self.measurements
+        ), "You have entered an invalid measurement, please choose one of {}.".format(
+            self.measurements
+        )
+        self.measurement = measurement
+
+        self.filename_glob = "*0_{}*[!zip]".format(self.measurement)
```

Comment:
I believe this matches any file containing `0_agb` or `0_gsv` that doesn't end with the letters z, i, or p? I don't think globs support what you're trying to do here.